apps/bridge/src/utils/hooks/usePrepareFinalizeWithdrawal.ts

```diff
@@ -4,44 +4,52 @@ import { WithdrawalMessage } from 'apps/bridge/src/types/WithdrawalMessage';
 import { getWithdrawalMessage } from 'apps/bridge/src/utils/transactions/getWithdrawalMessage';
 import getConfig from 'next/config';
 import { usePrepareContractWrite, useWaitForTransaction } from 'wagmi';
+import { useChainEnv } from 'apps/bridge/src/utils/hooks/useChainEnv';
+import { useTOSStatus } from 'apps/bridge/src/contexts/TOSContext';
 
 const { publicRuntimeConfig } = getConfig();
 
-export function usePrepareFinalizeWithdrawal(
-  withdrawalTx: `0x${string}`,
-  isERC20Withdrawal = false,
-) {
+export function usePrepareFinalizeWithdrawal(withdrawalTx: `0x${string}`) {
   const [withdrawalForTx, setWithdrawalForTx] = useState<WithdrawalMessage | null>(null);
+  const { isTosAccepted } = useTOSStatus();
+
+  const chainEnv = useChainEnv();
+  const isMainnet = chainEnv === 'mainnet';
+  const includeTosVersionByte = isMainnet;
+
   const { data: withdrawalReceipt } = useWaitForTransaction({
     hash: withdrawalTx,
     chainId: parseInt(publicRuntimeConfig.l2ChainID),
   });
 
+  const shouldPrepare = withdrawalForTx && (isTosAccepted || !isMainnet);
+
   const { config } = usePrepareContractWrite({
-    address: withdrawalForTx ? publicRuntimeConfig.l1OptimismPortalProxyAddress : undefined,
+    address: shouldPrepare ? publicRuntimeConfig.l1OptimismPortalProxyAddress : undefined,
     abi: OptimismPortal,
     functionName: 'finalizeWithdrawalTransaction',
     chainId: parseInt(publicRuntimeConfig.l1ChainID),
-    args: withdrawalForTx
+    args: shouldPrepare
       ? [
-        {
-          nonce: withdrawalForTx.nonce,
-          sender: withdrawalForTx.sender,
-          target: withdrawalForTx.target,
-          value: withdrawalForTx.value,
-          gasLimit: withdrawalForTx.gasLimit,
-          data: withdrawalForTx.data,
-        },
-      ]
+          {
```

Comment:
Is there any reason not to just pass in `withdrawalForTx` here? It seems like we just read all the same keys into a new object? If it needs to be a new object for some reason could we do `{...withdrawalForTx}`?

Reply:
hm yeah not sure why this was set up like this. can follow up & clean up